tests/testthat/test-iterative_solver.R

```diff
@@ -61,30 +54,23 @@ test_that("Iterative solver returns correct answer", {
   psusc <- matrix(1, nrow = 2, ncol = 1)
   susc <- psusc
 
-  # needed to get demography-risk combinations
-  epi_spread_data <- epi_spread(
+  epi_outcome <- final_size(
     contact_matrix = contact_matrix,
     demography_vector = demography_vector,
-    p_susceptibility = psusc,
-    susceptibility = susc
+    susceptibility = susc,
+    p_susceptibility = psusc
   )
 
-  epi_outcome <- solve_final_size_iterative(
-    contact_matrix = epi_spread_data[["contact_matrix"]],
-    demography_vector = epi_spread_data[["demography_vector"]],
-    susceptibility = epi_spread_data[["susceptibility"]]
-  )
-
-  epi_outcome_known <- 1 - exp(-r0 * epi_outcome)
+  epi_outcome_known <- 1 - exp(-r0 * epi_outcome$p_infected)
 
   tolerance <- 1e-5
   expect_true(
-    all(abs(epi_outcome - epi_outcome_known) < tolerance)
+    all(abs(epi_outcome$p_infected - epi_outcome_known) < tolerance)
```

Comment:
Could change to `expect_equal()` with tolerance set in the function argument. It might improve failure messages as it uses `waldo::compare()`.

Reply:
Now using `expect_equal`: https://github.com/epiverse-trace/finalsize/pull/73/commits/265c5469793bc9683f17f279ae68876a07b8a7c2